### PR TITLE
fix: remove the folder and update packages

### PR DESCRIPTION
**Pull request by Maledong**
1) Bug Fix: We should remove the folder with all of its files when re-running the tool, because if we don't want some files but they are created before, the tool won't automatically delete them. This will cause a problem for duplicated files left.

2) Update some packages, for details please see:
 a) 'debug': https://github.com/visionmedia/debug/releases/tag/4.1.1
 b) 'gh-pages': https://github.com/tschaub/gh-pages/releases/tag/v2.0.0
 c) 'hexo-renderer-less': https://github.com/hexojs/hexo-renderer-less/releases/tag/1.0.0

### CODE REVIEW

File: lib/base.js [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 const path = require('path');
 const fs = require('mz/fs');
+const fsExtra = require('fs-extra');
```
Comment (atian25) [inline lib/base.js]: use https://www.npmjs.com/package/mz-modules

Comment (atian25) [reply lib/base.js]:
```suggestion
const { rimraf } = require('mz-modules');
```

File: lib/base.js [hunk 2]
```diff
@@ -38,6 +39,24 @@ class DocToolsBase extends Command {
...
+
+    // If we've already had this folder, remove it
+    // 1. There're some files useless and they will be left if we don't remove the folder
+    // 2. Sometimes when we remove some folders in the 'run/doctools' and we directly run
+    // 'npm run test', the original files won't be removed this will make us make mistakes
+    // and we won't remove the related links, this will cause the submit error and the local
+    // test cannot be checked out.
+    const removePath = path.join(argv.baseDir, 'run');
+    try {
+      fsExtra.removeSync(removePath);
```
Comment (atian25) [inline lib/base.js]:
```suggestion
      yield rimraf(removePath);
```

**Comment by atian25** [comment]
commit message should start with lowercase `fix: `

File: lib/base.js [hunk 3]
```diff
@@ -38,6 +38,24 @@ class DocToolsBase extends Command {
...
+    // 2. Sometimes when we remove some folders in the 'run/doctools' and we directly run
+    // 'npm run test', the original files won't be removed this will make us make mistakes
+    // and we won't remove the related links, this will cause the submit error and the local
+    // test cannot be checked out.
+    const removePath = path.join(argv.baseDir, 'run');
+    try {
+      yield rimraf(removePath);
+      this.logger.info(`'${removePath}' is successfully removed, preparing to re-generate folders and files`);
+    } catch (e) {
+      this.logger.warn(`'${removePath}' failed to be removed,
```
Comment (atian25) [inline lib/base.js]: most time, we don't need to catch `rimraf`.

Comment (Maledong) [reply lib/base.js]: In order to avoid making mistakes, this is a safer way to notify you ;)

**Comment by Maledong** [comment]
@thonatos: Please have a review, thanks!

**Comment by thonatos** [comment]
Please rebase master, 😄

**Pull request merged**